api/proto/teleport/okta/v1/okta_service.proto

```diff
@@ -51,8 +51,166 @@ service OktaService {
   rpc DeleteOktaAssignment(DeleteOktaAssignmentRequest) returns (google.protobuf.Empty);
   // DeleteAllOktaAssignments removes all Okta assignments.
   rpc DeleteAllOktaAssignments(DeleteAllOktaAssignmentsRequest) returns (google.protobuf.Empty);
+
+  // ValidateClientCredentials checks if the provided client credentials are valid.
+  rpc ValidateClientCredentials(ValidateClientCredentialsRequest) returns (ValidateClientCredentialsResponse);
+  // EnrollIntegration enrolls a new integration with the specified parameters.
+  rpc EnrollIntegration(EnrollIntegrationRequest) returns (EnrollIntegrationResponse);
+  // UpdateIntegration updates the settings or properties of an existing integration.
+  rpc UpdateIntegration(UpdateIntegrationRequest) returns (UpdateIntegrationResponse);
+  // FilterApps retrieves a list of apps based on specified filter criteria.
+  rpc FilterApps(FilterAppsRequest) returns (FilterAppsResponse);
+  // FilterGroups retrieves a list of groups based on specified filter criteria.
+  rpc FilterGroups(FilterGroupsRequest) returns (FilterGroupsResponse);
+}
+
+// FilterAppsRequest is the request message for filtering apps.
+message FilterAppsRequest {
+  // URL of the Okta organization.
+  string okta_organization_url = 1;
+  // Credentials to access the Okta API.
+  OktaAPICredentials api_credentials = 2;
```

Comment:
In a given Oka sync service runtime, will there be just one credential or multiple credentials? I would omit trailing `s` if its the former.

Reply:
In our API we are referring to `Credentials` even if this is a single object: for instance: https://github.com/gravitational/teleport/blob/bd4b97a6af5ea7947985292bac1735d990471d3a/api/proto/teleport/legacy/types/types.proto#L6648

I don't have strong feeling but we should make it consistent